src/generator.jl

```diff
@@ -2,73 +2,81 @@ for (mv, sv, mm, sm, T) in ((:mkl_scscmv_, :mkl_scscsv_, :mkl_scscmm_, :mkl_scsc
                             (:mkl_dcscmv_, :mkl_dcscsv_, :mkl_dcscmm_, :mkl_dcscsm_, :Float64),
                             (:mkl_ccscmv_, :mkl_ccscsv_, :mkl_ccscmm_, :mkl_ccscsm_, :Complex64),
                             (:mkl_zcscmv_, :mkl_zcscsv_, :mkl_zcscmm_, :mkl_zcscsm_, :Complex128))
-    @eval begin
-        function cscmv!(transa::Char, α::$T, matdescra::ASCIIString, A::SparseMatrixCSC{$T, BlasInt}, x::StridedVector{$T}, β::$T, y::StridedVector{$T})
-            trns = uppercase(transa)
-            in(trns,['N','T']) || error("uppercase(transa) is '$trns', must be 'N' or 'T'")
-            length(x) == (trns == 'T' ? A.m : A.n) ||
-                throw(DimensionMismatch("Matrix with $(A.n) columns multiplied with vector of length $(length(x))"))
-            length(y) == (trns == 'T' ? A.n : A.m) ||
-                throw(DimensionMismatch("Vector of length $(A.m) added to vector of length $(length(y))"))
-            ccall(($(string(mv)), :libmkl_rt), Void,
-                (Ptr{Uint8}, Ptr{BlasInt}, Ptr{BlasInt}, Ptr{$T},
-                 Ptr{Uint8}, Ptr{$T}, Ptr{BlasInt}, Ptr{BlasInt},
-                 Ptr{BlasInt}, Ptr{$T}, Ptr{$T}, Ptr{$T}),
-                &transa, &A.m, &A.n, &α,
-                matdescra, A.nzval, A.rowval, pointer(A.colptr, 1),
-                pointer(A.colptr, 2), x, &β, y)
-            return y
-        end
+@eval begin
+function cscmv!(transa::Char, α::$T, matdescra::ASCIIString,
+                A::SparseMatrixCSC{$T, BlasInt}, x::StridedVector{$T},
+                β::$T, y::StridedVector{$T})
+    trns = uppercase(transa)
+    in(trns,['N','T']) || error("uppercase(transa) is '$trns', must be 'N' or 'T'")
```

Comment:
throw and argument error?  also `trns == 'N' || trns == 'T' || throw(ArgumentError(...))` would get rid of allocating an array just to check two values